MSAL/src/configuration/external/ios/NSString+MSALAccountIdenfiers.m

```diff
@@ -51,33 +45,19 @@ - (NSData *)msalStringAsGUIDData
     
     if (stringLen > 16)
     {
+        MSID_LOG_WITH_CTX(MSIDLogLevelWarning, nil, @"Failed to parse unsigned long long from a string, length: %d", stringLen);
         return nil;
     }
     
-    NSUInteger zeroFillLen = (16 - (stringLen + 1) / 2);
-    NSMutableData *result = [[NSMutableData alloc] initWithLength:16];
-    
-    char chars[3] = {'\0','\0','\0'};
-    
-    for (int i = stringLen-1; i >= 0; i-=2)
+    NSScanner *scanner = [NSScanner scannerWithString:self];
+    unsigned long long ull;
+    if (![scanner scanHexLongLong:&ull])
```

Comment:
For my learning, why do we use scanHexLongLong and not scanHexInt. Is it an unsigned long long value we are reading from the string ?

Reply:
The value won't fit into Int, ests sends the long long value to us.

Reply:
Thank you Serhii